app/scripts/controllers/transactions/index.js

```diff
@@ -928,6 +940,64 @@ export default class TransactionController extends EventEmitter {
     }
   }
 
+  async approveTransactionsWithSameNonce(listOfTxParams) {
+    // This is hacky, need to review with other engineers and decide on best alternative.
+    const uniqueHashOfParams = hashObject(listOfTxParams);
```

Comment:
Still need to figure out the best alternative to this:

- [ ] how to determine if transaction is in process of signing

Reply:
Let us please know if you figure it out or we can have a quick brainstorming session to find the best solution.

Reply:
Maybe we could just get a hash of the first transaction in the array, using TransactionFactory... that would be a bit better I think

Reply:
but we might not be able to do that for an unsigned transaction...

Reply:
I pushed a commit here that I think resolves my concerns: https://github.com/MetaMask/metamask-extension/commit/54a01beabb0e0d15cf451584982f32923834dddb